models/favourite.models.js

```diff
@@ -0,0 +1,60 @@
+const db = require("../db/connection");
+const { fetchUserByUserId } = require("./user.models");
+const { fetchClothByClothesId } = require("./clothes.models");
+
+async function fetchFavourites(user_id) {
+    await fetchUserByUserId(user_id);
+    
+	const sqlString = `SELECT * FROM favourites WHERE uid = $1`;
+
+	const { rows: favourites } = await db.query(sqlString, [user_id]);
+
+	return favourites;
+}
+
+async function addFavourite(user_id, newFavourite) {
+    try {
+        console.log(user_id);
+        console.log(newFavourite.clothes_id);
+        
+        if (!newFavourite.clothes_id) {
+            throw { 
+                status: 400, 
+                msg: "Bad request!" 
+            };
+        }
+        if(typeof newFavourite.clothes_id !== "number") {
+            return Promise.reject({
```

Comment:
does throw not work here?

Reply:
nice catch, fixed it!

Reply:
sorry i didnt mean it was wrong or a problem, just under some circumstances you cant throw an error it has to be promise rejected, i think its only when the throw is nested in a promise though